scripts/pipeline/set_environment_variables/set_slack_post_environment_domains.sh

```diff
@@ -1,63 +1,68 @@
 #! /bin/bash
+    # needed as circleci did not santize the input for json properly.
+    SANITISED_COMMIT_MESSAGE=$(echo "${COMMIT_MESSAGE//$'\n'/\\\n}"  | sed 's/"/\\""/g')
```

Comment:
Could you please give me a brief explanation of what's going on here as it's a bit opaque? I get the sed part but I'm not sure what's going on with the echo statement. Just for my own understanding, not a block on merging it.

Reply:
The `echo` writes out to the shell variable with a substitution of actual new lines with a new line literal e.g. `\n'. Think of it as like sed command similar to `sed 's//$'\n'/\\\n/g`.

Reply:
To be fair it was a first attempt so could make this a bit clearer...